dotcom-rendering/src/web/browser/hydration/getName.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * getName takes the given html element and returns its name attibute
+ *
+ * We expect the element to always be a `gu-*` custom element
```

Comment:
For future reference (and for reviewers in a hurry!) what is the role and significance of the `gu-*` custom element?

Reply:
They're [cooler](https://github.com/guardian/dotcom-rendering/pull/445#pullrequestreview-209189589) 😉 

Maybe they also provide a nice separation from standard elements as a way to highlight these markers as not being part of the semantic dom structure?

Reply:
Okay thanks for clarifying that, I agree it flags up the hydratable markup better than a class. I'm wondering if the naming scales well with some possible future in which many of our components are custom elements. Do you think we should choose a name that implies these elements need to be hydrated on the client?

Reply:
> I'm wondering if the naming scales well with some possible future in which many of our components are custom elements

Good point. It's worth pointing out I had in mind using `gu-hydrate` + `gu-portal` at some point soon, hence the asterisk

Reply:
_Very slight_ niggle that this is overloading custom elements syntax when they are not _really_ custom elements but rehydrated Preact components (iiuc).

But it is cooler.